src/library-authoring/create-collection/CreateCollectionModal.tsx

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import {
+  ActionRow,
+  Button,
+  Form,
+  ModalDialog,
+} from '@openedx/paragon';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import { useParams } from 'react-router-dom';
+import { LibraryContext } from '../common/context';
+import messages from './messages';
+import { useCreateLibraryCollection } from '../data/apiHooks';
+import { ToastContext } from '../../generic/toast-context';
+
+const CreateCollectionModal = () => {
+  const intl = useIntl();
+  const { libraryId } = useParams();
+  const create = useCreateLibraryCollection(libraryId);
+  const {
+    isCreateCollectionModalOpen,
+    closeCreateCollectionModal,
+  } = React.useContext(LibraryContext);
+  const { showToast } = React.useContext(ToastContext);
+
+  const [collectionName, setCollectionName] = React.useState<string | null>(null);
```

Comment:
We should not use `null` as a value on an input. It should be `''` for a controlled component. This generates some warnings in the console.
```suggestion
  const [collectionName, setCollectionName] = React.useState<string>('');
```

Also, I think it would be better to use Formik + Yup for form validation and state management instead of doing it manually. We have a recent example here: https://github.com/openedx/frontend-app-course-authoring/blob/master/src/library-authoring/create-library/CreateLibrary.tsx

Reply:
Updated: https://github.com/openedx/frontend-app-course-authoring/pull/1259/commits/aa870b4b48ed8035549476d11681c619b0cd2cbe